aws/table_aws_shield_attack.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/shield"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsShieldAttack(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_shield_attack",
+		Description: "AWS Shield Attack",
+		List: &plugin.ListConfig{
+			Hydrate: listAwsShieldAttack,
```

Comment:
I was thinking of doing it this way and not create two tables (`aws_shield_attack` and `aws_shield_attack_summary`), however, if both would be combined, then one query against this table would trigger many `DescribeAttack` API requests. This would lead quicker to throttling. By separating bot APIs (`DescribeAttack` -> `aws_shield_attack` and `ListAttack` -> `aws_shield_attack_summary`), it would allow the user more flexibility to efficiently implement their queries without hitting the rate limit.

Reply:
From an API perspective, both `ListAttacks` and `DescribeAttack` return information about attacks, so there is no need to create separate tables using these APIs.

We should include both `ListConfig` and `GetConfig` in the `aws_shield_attack` table. To handle throttling errors, we can limit the results by setting the [MaxResults](https://docs.aws.amazon.com/waf/latest/DDOSAPIReference/API_ListAttacks.html#AWSShield-ListAttacks-request-MaxResults) parameter, and manage concurrency by setting `max_concurrency` in the Steampipe connection or as an environment variable. 

`aws_shield_attack` table can get a particular attack details by providing the `attack_id` in the WHERE clause.

Reply:
Alright, so in this case I will rearrange the tables a little bit, such `aws_shield_attack_summary` gets renamed to `aws_shield_attack` and the columns of the former `aws_shield_attack` are added to the list of columns of the new table.